source/backend/api/Services/DocumentFileService.cs

```diff
@@ -89,333 +91,509 @@ public IList<T> GetFileDocuments<T>(FileType fileType, long fileId)
+            var result = new ExternalResponse<string>() { Status = ExternalResponseStatus.NotExecuted };
+            PimsDocument currentDocument = _documentRepository.Find(dispositionFileDocument.DocumentId);
+
+            // 1 - Delete Mayan first.
+            if (currentDocument.MayanId.HasValue && currentDocument.MayanId.Value > 0)
             {
-                return await documentService.DeleteDocumentAsync(dispositionFileDocument.Document);
+                result = await DeleteMayanDocument((long)currentDocument.MayanId);
+                currentDocument = RemoveDocumentMayanID(currentDocument);
+                _documentRepository.CommitTransaction(); // leave trace when mayan document deleted.
             }
-            else
+
+            using var transaction = _documentRepository.BeginTransaction();
+
+            // 2 - Delete Queue Item
+            DeleteQueuedDocumentItem(currentDocument.DocumentId);
+
+            // 3 - Delete FileDocument
+            _dispositionFileDocumentRepository.DeleteDispositionDocument(dispositionFileDocument);
+
+            // 4 - Delete Document
+            DeleteDocument(currentDocument);
+
+            // All good
+            _documentRepository.SaveChanges();
+            transaction.Commit();
+
+            return result;
+        }
+
+        private PimsDocument CreatePimsDocument(DocumentUploadRequest uploadRequest, string documentExternalId = null)
+        {
+            // Create the pims document
+            PimsDocument newPimsDocument = new()
+            {
+                FileName = uploadRequest.File.FileName,
+                DocumentTypeId = uploadRequest.DocumentTypeId,
+                DocumentStatusTypeCode = uploadRequest.DocumentStatusCode,
+                MayanId = null,
+                DocumentExternalId = documentExternalId,
+            };
+
+            _documentRepository.Add(newPimsDocument);
+
+            return newPimsDocument;
+        }
+
+        private async Task<ExternalResponse<string>> DeleteMayanDocument(long mayanDocumentId)
+        {
+            var result = await _documentService.DeleteMayanStorageDocumentAsync(mayanDocumentId);
+            if (result.HttpStatusCode == HttpStatusCode.NotFound)
             {
-                _dispositionFileDocumentRepository.DeleteDispositionDocument(dispositionFileDocument);
-                _dispositionFileDocumentRepository.CommitTransaction();
-                return new ExternalResponse<string>() { Status = ExternalResponseStatus.NotExecuted };
+                result.Status = ExternalResponseStatus.Success;
             }
+
+            return result;
         }
 
-        private static void ValidateZeroLengthFile(DocumentUploadRequest uploadRequest)
+        private PimsDocument RemoveDocumentMayanID(PimsDocument doc)
         {
-            if (uploadRequest.File is not null && uploadRequest.File.Length == 0)
+            // Update the Mayan ID from PIMS, this will leave a trace
```

Comment:
I don't think this method executed alone will leave a trace.

Reply:
making sure that the mayan Id is deleted is the calling to mayan api is successful and making sure that mayan id is null after that so it does not point to a deleted file. trace is just the history table